### PR TITLE
slurm-wlm: use lazy bind (fix plugin loading)

### LINKED ISSUE #25123: slurm-wlm: Error relocating ... symbol not found
### System

* xuname:  
Void 5.8.10_1 x86_64-musl GenuineIntel notuptodate rrrnFFFFFF
Happens on glibc too (tried in a chroot)
* package:  slurm-wlm-19.05.5.1_2

### Expected behavior
No errors

### Actual behavior
```
slurmd: error: plugin_load_from_file: dlopen(/usr/lib/slurm/select_cons_res.so): Error relocating /usr/lib/slurm/select_cons_res.so: slurm_job_preempt_mode: symbol not found
slurmd: error: Couldn't load specified plugin name for select/cons_res: Dlopen of plugin file failed
slurmd: fatal: Can't find plugin for select/cons_res
```

### Steps to reproduce the behavior

1. Create a config via `/usr/share/doc/slurm-19.05.5/html/configurator.easy.html`
2. Launch `/bin/slurmd`

Other binaries give similar errors, for example `sacctmgr`:
```
sacctmgr: error: plugin_load_from_file: dlopen(/usr/lib/slurm/accounting_storage_slurmdbd.so): Error relocating /usr/lib/slurm/accounting_storage_slurmdbd.so: lock_slurmctld: symbol not found
sacctmgr: error: Couldn't load specified plugin name for accounting_storage/slurmdbd: Dlopen of plugin file failed
```

The package seems generally broken...

### PR DESCRIPTION
**Pull request by st3r4g**
fixes #25123
ping maintainer @Vaelatern

### CODE REVIEW

File: srcpkgs/slurm-wlm/template [hunk 1]
```diff
@@ -19,6 +19,7 @@ distfiles="https://github.com/SchedMD/slurm/archive/slurm-${_distver}.tar.gz"
 checksum=e53e67bd0bb4c37a9c481998764a746467a96bc41d6527569080514f36452c07
 
 CFLAGS="-fcommon"
+LDFLAGS="-Wl,-z,lazy"
```
Comment (ericonr) [inline srcpkgs/slurm-wlm/template:22]: `# fix plugin loading for musl`

Comment (st3r4g) [reply srcpkgs/slurm-wlm/template:22]: it's not just for musl. I'll add `# fix plugin loading`

**Comment by ericonr** [comment]
Do you think our build is overwriting upstream's `LDFLAGS`?

**Comment by Vaelatern** [comment]
An actual user for slurm-wlm!

Please add one comment saying "Yes, this disables hardening, this is on purpose" and link to the open bug? I am willing to merge once this necessary context is added.

**Comment by st3r4g** [comment]
> Do you think our build is overwriting upstream's `LDFLAGS`? Or are they failing to set that flag?

What do you mean? The problem is the `-z,now` default on Void

> An actual user for slurm-wlm!

I don't know how to use it. I just installed it to mess around with it a bit

**Comment by ericonr** [comment]
> What do you mean? The problem is the -z,now default on Void

I noticed later, sorry.

**Pull request merged**